sdk/java/src/main/java/com/pulumi/kubernetes/apiextensions/CustomResource.java

```diff
@@ -0,0 +1,220 @@
+// *** WARNING: this file was generated by pulumi-java-gen. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package com.pulumi.kubernetes.apiextensions;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import com.pulumi.core.Either;
+import com.pulumi.core.Output;
+import com.pulumi.core.annotations.Export;
+import com.pulumi.core.annotations.Import;
+import com.pulumi.core.annotations.ResourceType;
+import com.pulumi.core.internal.Codegen;
+import com.pulumi.core.internal.Internal;
+import com.pulumi.core.internal.OutputInternal;
+import com.pulumi.core.internal.annotations.ImportMetadata;
+import com.pulumi.kubernetes.Utilities;
+import com.pulumi.kubernetes.meta.v1.outputs.ObjectMeta;
+import com.pulumi.resources.ResourceArgs;
+
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.description.annotation.AnnotationDescription;
+import net.bytebuddy.implementation.FieldAccessor;
+import net.bytebuddy.implementation.Implementation;
+import net.bytebuddy.implementation.MethodCall;
+
+/**
+ * CustomResource represents an instance of a CustomResourceDefinition (CRD). For example, the
+ * CoreOS Prometheus operator exposes a CRD `monitoring.coreos.com/ServiceMonitor`; to
```

Comment:
Would it be practical to add an example in-tree, even if it maybe cannot fully run as an integration test, just to demonstrate usage?